tests/scenarios.py

```diff
@@ -112,7 +112,8 @@ def prepare(self):
         self.target = os.path.join(self.temp, "work")
 
         os.makedirs(self.origin)
-        self.run_git("init", "--bare", self.origin, cwd=self.temp)
+        # TODO: need to stop defaulting to 'master' branch, and use non-hardcoded git default instead ('main' nowadays)
```

Comment:
Why not look up the repo config for this? e.g. `git ls-remote --symref origin HEAD`

Reply:
Tried, but doesn't run anything in the test setup on a freshly `git init`-ed repo... Will have to ponder how to better do that in general

Reply:
Interesting. The above did require that you had cloned from a remote repository, since git doesn't really have a notion of a "local" primary branch.